src/app/services/EndpointService.js

```diff
@@ -0,0 +1,58 @@
+(function () {
+  'use strict';
+  var m = angular.module('nflowExplorer.services.EndpointService', [
+    'nflowExplorer.config',
+  ]);
+  m.service('EndpointService', function EndpointService(config, $window) {
+    var api = this;
+    api.currentEndpoint = currentEndpoint;
+    api.selectEndpoint = selectEndpoint;
+    api.availableEndpoints = availableEndpoints;
+    api.init = init;
+
+    function currentEndpoint() {
+      return config.selectedEndpoint;
+    }
+
+    function updateEndpoint(endpoint) {
+      if(!endpoint) {
+        return;
+      }
+      var change = config.selectedEndpoint && config.selectedEndpoint.id !== endpoint.id;
+      config.selectedEndpoint = endpoint;
+      config.nflowUrl = endpoint.apiUrl;
+      config.nflowApiDocs = endpoint.docUrl;
+
+      try {
+        $window.localStorage.setItem('nflow.endpoint', endpoint.id);
+      } catch (e) {}
+
+      if(change) {
+        $window.location.reload();
```

Comment:
This reload fails, if you have opened a workflow definition or workflow instance (e.g. http://localhost:9000/#/workflow/4) that does not exist in the new endpoint. 
Suggestions:
1) workflow definitions: add error page that notifies "Workflow definition <type> does not exist on <endpoint name>." (for example) and has link to workflow definitions listing page. Alternative redirect directly to listing page and put the notification on top of the page.
2) workflow instances: if workflow instance is selected, go directly to "search workflow instances" page. It doesn't serve any purpose if random other workflow instance is shown when switching endpoint.

There could also be some notification "Changed to nFlow endpoint <endpoint name>" etc if it is easy to do.